homeassistant/helpers/llm.py

```diff
@@ -0,0 +1,111 @@
+"""Module to coordinate llm tools."""
+
+from __future__ import annotations
+
+from abc import abstractmethod
+from collections.abc import Iterable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.weather.intent import INTENT_GET_WEATHER
+from homeassistant.core import Context, HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+
+from . import intent
+
+_LOGGER = logging.getLogger(__name__)
+
+IGNORE_INTENTS = [intent.INTENT_NEVERMIND, intent.INTENT_GET_STATE, INTENT_GET_WEATHER]
+
+
+@dataclass(slots=True)
+class ToolInput:
+    """Tool input to be processed."""
+
+    tool_name: str
+    tool_args: dict[str, Any]
+    platform: str
+    context: Context | None
+    user_prompt: str | None
+    language: str | None
+    agent_id: str | None
+    conversation_id: str | None
+    device_id: str | None
```

Comment:
This appears to be unused

Reply:
It is for future use. I can imagine an LLM tool where it is used.
While it is indeed not used in intents for now, I can remove it and add it later when we add support for tool registration, but eventually I believe it should be there. Do you want me to do it?

Same comment for `agent_id` and `conversation_id`.

Reply:
Yeah, we should not include anything right now that we're not using. Focus on MVP. Too much in flux and we need to first see if this approach can work and how it will evolve.

Reply:
Done